torch/nn/parallel/distributed.py

```diff
@@ -665,6 +668,27 @@ def _sync_params_and_buffers(self, authoritative_rank=0):
                 module_states, self.broadcast_bucket_size, authoritative_rank
             )
 
+    def _mark_params_and_buffers_as_replicated(self, buffers_only=False):
+        from torch.distributed._shard.replicated_tensor import ReplicatedTensor, ReplicatedParameter
+        for module_name, module in self.module.named_modules():
+            if not buffers_only:
+                for param_name, param in module.named_parameters(recurse=False):
+                    if (param.requires_grad and f'{module_name}.{param_name}'
+                            not in self.parameters_to_ignore and not isinstance(param, ReplicatedParameter)):
```

Comment:
does the latter check mean that it is possible to call this function on already replicated parameters? Should we try to make it so that the application doesn't do this because it would be unnecessary?